### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by kryztoval** [aeternity/hd-wallet-js#4]
I did all the steps in the beginning, and I am not sure what to expect or how to see it. :(

**Comment by davidyuk** [comment]
This package exposes two functions: `generateHDWallet` and `getHDWalletAccounts`, they are defined in [this file](https://github.com/aeternity/hd-wallet-js/blob/474915eb81279ed800ff1266efc2513c54976d16/src/hd-wallet.js).

In the Base app, `generateHDWallet` is [used to generate wallet](https://github.com/aeternity/aepp-base/blob/ca0c8657167c00a1ec4a42a787480a13fe5b6530/src/store/modules/mobile.js#L107) by mnemonic phrase. `getHDWalletAccounts` function is [called every time](https://github.com/aeternity/aepp-base/blob/219f2180d7bfb37a9c5e0f073500dbd613a4350f/src/store/plugins/decryptAccounts.js#L14) when changes one of `keystore`, `accountCount`, or `derivedKey`.

Feel free to ask any additional questions if you have.

**Comment by kryztoval** [comment]
Does that mean I have to use this as a backend, and code a front end to use it?

**Comment by davidyuk** [comment]
This is not a complete application, this package exposes functions for the derivation of aeternity accounts. This functions can be used in both browser and nodejs.

**Comment by kryztoval** [comment]
Is there a full wallet? even a console wallet that can sign messages, sign transactions or something like that somewhere? Sorry if I ask too many questions.

I would suggest adding a "What hd-wallet is" in the readme.md explaining it is a library. 😄

**Comment by davidyuk** [comment]
> Is there a full wallet?

No, this package implements only the derivation of aeternity accounts, firstly I propose to make it a part of `aepp-sdk-js`, but we decided to extract it into a separate package: https://github.com/aeternity/aepp-sdk-js/pull/41

> even a console wallet that can sign messages, sign transactions or something like that somewhere?

please, check this repository: https://github.com/aeternity/aepp-sdk-js/

> I would suggest adding a "What hd-wallet is" in the readme.md explaining it is a library

I agree that we have a lack of documentation in this repository.

**Comment by kryztoval** [comment]
I know I am new here and that I don't yet understand everything that there is around this repository, thanks for the patience!

**Comment by davidyuk** [comment]
Let's keep this issue opened until we add some documentation to this repository.
